src/cpp/fastdds/xtypes/dynamic_types/idl_parser/IdlParser.hpp

```diff
@@ -287,6 +353,38 @@ struct action<scoped_name>
                 throw std::runtime_error("Unknown constant or identifier: " + identifier_name);
             }
         }
+        else if (state["type"] == "sequence" && state["element_type"].empty())
+        {
+            // <scoped_name> is the element type of a sequence (previously declared)
+            state["element_type"] = identifier_name;
+
+            // Ready to parse the sequence size (if provided)
+            state["arithmetic_expr"] = "";
+        }
+        else if (state.count("alias"))
+        {
+            // <scoped_name> makes reference to the aliased type.
+            //
+            // There are two possible cases:
+            // 1. The aliased type is a custom and previously declared type.
+            //    In this case, <scoped_name> is the aliased type name and should be stored in state["alias"].
+            //
+            // 2. The aliased type is not a previously declared type
+            //    (i.e: typedef <type> <alias>, when <type> is a Primitive type, string type, sequence type or map type
+            //    not previously declared), it contains internal types (e.g: the type of elements in a sequence),
+            //    and these types are previosly declared by the user,
+            //    i.e: <type>:=external_type<internal_type_1,...>, when <internal_type_i> is a previously declared type.
+            //    It can occur when external_type is a sequence or a map. In this case, the identifier_name matched correspond
+            //    to an internal type name, should be stored in specific states keys (leaving state["alias"] empty)
+            //    and must be handled depending on state["type"].
+            //
+            //    In case of sequence types, case 2 is possible but already handled in the previous else if statement.
+            state["alias"] = identifier_name;
+        }
```

Comment:
Here the second case is the previous 'else if (state["type"] == "sequence" && state["element_type"].empty())' so doesn't make to much sense to explain the case here again.

```suggestion
        else if (state.count("alias"))
        {
            // <scoped_name> makes reference to the aliased type.
            //
            //    The aliased type is a custom and previously declared type.
            //    In this case, <scoped_name> is the aliased type name and should be stored in state["alias"].
            state["alias"] = identifier_name;
        }
```